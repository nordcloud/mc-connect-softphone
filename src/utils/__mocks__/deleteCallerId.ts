export const deleteCallerId = jest.fn().mockResolvedValue('dummyCallerId');
