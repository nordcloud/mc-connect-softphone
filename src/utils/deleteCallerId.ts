import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
import { DeleteCommand, DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';

import { CALLER_ID_TABLE } from '../consts';

const ddbClient = new DynamoDBClient({});
const ddbDocClient = DynamoDBDocumentClient.from(ddbClient);

export async function deleteCallerId(username: string) {
  const deleteCommand = new DeleteCommand({
    TableName: CALLER_ID_TABLE,
    Key: { username },
    ReturnValues: 'ALL_OLD',
  });

  const { Attributes } = await ddbDocClient.send(deleteCommand);

  return Attributes?.callerId;
}
