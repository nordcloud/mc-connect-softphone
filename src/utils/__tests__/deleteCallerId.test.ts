import { DeleteCommand, DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';
import { mockClient } from 'aws-sdk-client-mock';

import { deleteCallerId } from '../deleteCallerId';

const ddbMock = mockClient(DynamoDBDocumentClient);

const args = ['username'] as const;

beforeEach(() => {
  ddbMock.reset();
  ddbMock.on(DeleteCommand).resolves({ Attributes: { callerId: 'dummyCallerId' } });
});

describe('deleteCallerId', () => {
  it('sends "DeleteCommand" with correct args', async () => {
    await deleteCallerId(...args);
    expect(ddbMock.commandCalls(DeleteCommand)[0].args[0].input).toMatchSnapshot();
  });

  it('resolves with a correct value', () => {
    return expect(deleteCallerId(...args)).resolves.toMatchSnapshot();
  });

  it('resolves with a correct value when output from DeleteCommand is empty', () => {
    ddbMock.on(DeleteCommand).resolvesOnce({});
    return expect(deleteCallerId(...args)).resolves.toMatchSnapshot();
  });
});
