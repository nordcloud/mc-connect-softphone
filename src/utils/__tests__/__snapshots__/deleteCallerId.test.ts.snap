// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteCallerId resolves with a correct value 1`] = `"dummyCallerId"`;

exports[`deleteCallerId resolves with a correct value when output from DeleteCommand is empty 1`] = `undefined`;

exports[`deleteCallerId sends "DeleteCommand" with correct args 1`] = `
{
  "Key": {
    "username": "username",
  },
  "ReturnValues": "ALL_OLD",
  "TableName": "mc-connect-softphone-prod-agents-caller-id",
}
`;
